docs/src/callbacks.md

```diff
@@ -0,0 +1,168 @@
+```@meta
+CurrentModule = JuMP
+DocTestSetup = quote
+    using JuMP
+end
+DocTestFilters = [r"≤|<=", r"≥|>=", r" == | = ", r" ∈ | in ", r"MathOptInterface|MOI"]
+```
+
+# Callbacks
+
+Many mixed-integer (linear, conic, and nonlinear) programming solvers offer
+the ability to modify the solve process. Examples include changing branching
+decisions in branch-and-bound, adding custom cutting planes, providing custom
+heuristics to find feasible solutions, or implementing on-demand separators to
+add new constraints only when they are violated by the current solution (also
+known as lazy constraints).
+
+While historically this functionality has been limited to solver-specific
+interfaces, JuMP provides solver-independent support for three types of
+callbacks:
+
+ 1. lazy constraints
+ 2. user-cuts
+ 3. heuristic solutions
+
+## Available solvers
+
+Callback support is limited to a few solvers. This includes
+[CPLEX](https://github.com/JuliaOpt/CPLEX.jl),
+[GLPK](https://github.com/JuliaOpt/GLPK.jl), and
+[Gurobi](https://github.com/JuliaOpt/Gurobi.jl).
+
+!!! warning
+    While JuMP provides a solver-independent way of accessing callbacks, the
+    particular implementations are dependent on the solver. You should not use
+    the callbacks without understanding how your particular solver handles
+    each callback.
+
+## Information that can be queried during callbacks
+
+In a callback, the only thing you may query is the primal value of the
+variables using [`callback_value`](@ref).
+
+If you need any other information, use a solver-dependent callback instead.
```

Comment:
What about some hints on how to use solver-dependent callbacks? (Or if they're not ready/documented, say so.)